lib/dialects/mssql/connection-manager.js

```diff
@@ -69,9 +69,18 @@ class ConnectionManager extends AbstractConnectionManager {
 
       const connection = new this.lib.Connection(connectionConfig);
       const connectionLock = new ResourceLock(connection);
+      let connected = false;
       connection.lib = this.lib;
 
+      connection.on('end', () => {
+        if (!connected) {
+          reject(new sequelizeErrors.ConnectionError('Error connecting to the database'));
```

Comment:
This is something acceptable, but this is not the actual error. Its a timeout, which is already returned by pooling library? (It should be, as Resource Request Timeout). May be we should return a `SequelizeTimeoutError` or something

Reply:
I am happy to help to fix that ASAP.
Can I, somehow, edit the PR to do those changes?

Thanks :)

Reply:
@cladjules do you think it's possible for me to just transfer the entire repo to your name?

Reply:
@iamjoyce  let me try to do a new PR and copy your code manually, there is not much to change. Thanks.

Reply:
@cladjules did you get a chance to look at this? Instead of copying the code you should be able to branch off @iamjoyce branch and then do a PR onto @iamjoyce or open a new PR directly on here.

Reply:
I will try to have a look this weekend. But quite busy. I will see what I can do and how I can reproduce...